### PR TITLE
[Dashboard] Capabilities page

### PR DESCRIPTION
**Pull request by sunny0826**
Capabilities page #740

### Workload

- list WorkloadDefinitions
- get details of WorkloadDefinition

### Operations

- list TraitDefinitions 
- get TraitDefinition

### CODE REVIEW

File: dashboard/config/routes.ts
Comment (hongchaodeng) [inline dashboard/config/routes.ts:15]:
```suggestion
    path: '/capabilities',
```

File: dashboard/config/routes.ts
Comment (hongchaodeng) [inline dashboard/config/routes.ts:19]:
```suggestion
        redirect: `/capabilities/workloads`,
```

File: dashboard/config/routes.ts
Comment (hongchaodeng) [inline dashboard/config/routes.ts:23]: pay attention to the capitalization here

**Comment by hongchaodeng** [comment]
@majian159 Can you review this PR?

File: dashboard/config/routes.ts
Comment (hongchaodeng) [inline dashboard/config/routes.ts:27]: Still call this "traits"?
Don't introduce new concepts since users might be confused.

**Review by hongchaodeng — APPROVED** [review]
LGTM in general. Nice job!

defer to @majian159 for another review.

File: dashboard/config/routes.ts
Comment (sunny0826) [reply dashboard/config/routes.ts:27]: @zzxwill According to ISSUE #740  content "operating" is used here, is it changed back to "traits"?

Comment (zzxwill) [reply dashboard/config/routes.ts:27]: @sunny0826 Let's take hongchao's advice. Thanks.

**Review by zzxwill — APPROVED** [review]
LGTM

File: dashboard/config/routes.ts
Comment (sunny0826) [reply dashboard/config/routes.ts]: fixed

**Comment by wonderflow** [comment]
please squash before merge

**Review by majian159 — CHANGES_REQUESTED** [review]

File: dashboard/src/pages/Capability/Components/ShowComponent/index.tsx
Comment (sunny0826) [reply dashboard/src/pages/Capability/Components/ShowComponent/index.tsx:1]: I don't understand what you mean, these code formats are from the IDE `Reformat Code`

image

File: dashboard/config/routes.ts
Comment (sunny0826) [reply dashboard/config/routes.ts:24]: yep

File: dashboard/src/pages/Capability/Components/ShowComponent/index.tsx
Comment (majian159) [reply dashboard/src/pages/Capability/Components/ShowComponent/index.tsx:1]: Please exec 'yarn lint' does not error.
I fixed 'lint index.less error', please wait merge and rebase

Comment (majian159) [reply dashboard/src/pages/Capability/Components/ShowComponent/index.tsx:1]: use exec `yarn prettier`, keep the project unified?

Comment (sunny0826) [reply dashboard/src/pages/Capability/Components/ShowComponent/index.tsx]: fixed

File: dashboard/src/pages/Capability/Components/ShowComponent/index.tsx
Comment (majian159) [inline dashboard/src/pages/Capability/Components/ShowComponent/index.tsx:35]: Please set a unique key.  
fix `Warning: Each child in a list should have a unique "key" prop.`

File: dashboard/src/pages/Capability/Components/ShowComponent/index.tsx
Comment (majian159) [inline dashboard/src/pages/Capability/Components/ShowComponent/index.tsx:58]: used `Tooltip` or set `showTitle: true` ?

File: dashboard/src/pages/Capability/Components/ShowComponent/index.tsx
Comment (majian159) [inline dashboard/src/pages/Capability/Components/ShowComponent/index.tsx:76]: text is empty not return <Text>, return null?

WeChat15bf5f38334deb2722ea3e8bb718a1aa

**Review by majian159 — CHANGES_REQUESTED** [review]
fix repeated clicks are invalid.

QQ20201225-111219-HD

**Comment by sunny0826** [comment]
I will fix these problems

File: dashboard/src/pages/Capability/Components/ShowComponent/index.tsx
Comment (sunny0826) [reply dashboard/src/pages/Capability/Components/ShowComponent/index.tsx:58]: Tooltip is already in use

image

**Pull request merged**
